x/stakeibc/keeper/hooks.go

```diff
@@ -53,20 +54,56 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 				}
 			}
 		}
+
+		// DELEGATE FROM DELEGATION ACCOUNT
 		delegateInterval := int64(k.GetParam(ctx, types.KeyDelegateInterval))
 		if epochNumber%delegateInterval == 0 {
+			// get Gaia LC height
 			k.ProcessDelegationStaking(ctx)
 		}
+
 		exchangeRateInterval := int64(k.GetParam(ctx, types.KeyExchangeRateInterval))
-		if epochNumber%exchangeRateInterval == 3 { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateDelegatedBalance(ctx)
-		}
-		if epochNumber%exchangeRateInterval == 6 && (epochNumber > 100) { // allow a few blocks from UpdateDelegatedBal to avoid conflicts & wait until chain has registered zones to calc exch rate
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateExchangeRate(ctx)
-		}
+		if epochNumber%exchangeRateInterval == 0 && (epochNumber > 100) { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
```

Comment:
agreed, we should wrap this into a function in the keeper which is called here 

why wait until 100 epochs have passed? is this to allow the channels / connections to be setup properly?

Reply:
yes it allows setup